src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,140 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeption.UserInvalidExeption;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    public static final String VALID_LOGIN = "John_Doe";
+    public static final String VALID_USERPASSWORD = "strongPassword";
+    public static final String SHORT_LOGIN = "Joe";
+    public static final String SHORT_PASSWORD = "abcde";
+    public static final int VALID_USERAGE = 25;
+    private RegistrationServiceImpl registrationService;
+    private StorageDaoImpl storageDao;
+    private User validUser;
+
+    @BeforeEach
+    void setUp() {
+        validUser = new User();
+        validUser.setAge(VALID_USERAGE);
+        validUser.setLogin(VALID_LOGIN);
+        validUser.setPassword(VALID_USERPASSWORD);
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_ValidUser_Ok() {
+        registrationService.register(validUser);
+        assertEquals(validUser, storageDao.get(validUser.getLogin()));
+    }
+
+    @Test
+    void register_UserWithNullLogin_Not_Ok() {
+        validUser.setLogin(null);
+
+        assertThrows(UserInvalidExeption.class, () ->
+                registrationService.register(validUser));
+        UserInvalidExeption userInvalidExeption = assertThrows(UserInvalidExeption.class, () ->
+                registrationService.register(validUser));
+        assertEquals("Please enter the login!", userInvalidExeption.getMessage());
+    }
+
+    @Test
+    void register_UserWithShortLogin_Not_Ok() {
+        validUser.setLogin(SHORT_LOGIN);
+
+        assertThrows(UserInvalidExeption.class, () ->
+                registrationService.register(validUser));
+        UserInvalidExeption userInvalidExeption = assertThrows(UserInvalidExeption.class, () ->
+                registrationService.register(validUser));
+        assertEquals("Login " + validUser.getLogin() + " must be at least 6 characters!",
+                userInvalidExeption.getMessage());
+    }
+
+    @Test
+    void register_UserWithExistingLogin_Not_ok() {
+        User userWithExistingLogin = new User();
+        userWithExistingLogin.setAge(38);
+        userWithExistingLogin.setPassword(VALID_USERPASSWORD);
+        userWithExistingLogin.setLogin(VALID_LOGIN);
+
+        assertThrows(UserInvalidExeption.class, () ->
+                registrationService.register(userWithExistingLogin));
```

Comment:
I can see that you do not put the first user in the storage here, and using the one that you have registered in the `register_ValidUser_Ok()`, but that shows us that our tests are not isolated from each other, when they should actually be. We can achieve that by clearing the `Storage.people` after/before each test, after that we can put the first user in this test just using the `storage` or `storageDao` instead because we can not guarantee the proper work of register method yet, to be able to use it for test data arrangement(see common mistakes for more info on that matter)

Reply:
Not understand all, but will ask at QnA.
Tnx for advice.